modules/message-lane/src/instant_payments.rs

```diff
@@ -15,57 +15,80 @@
-/// 'relayers-fund' account.
-pub struct InstantCurrencyPayments<AccountId, Currency> {
-	_phantom: sp_std::marker::PhantomData<(AccountId, Currency)>,
+/// Instant message payments made in given currency.
+///
+/// The balance is initally reserved in a special `relayers-fund` account, and transferred
+/// to the relayer when message delivery is confirmed.
+///
+/// NOTE The `relayers-fund` account must always exist i.e. be over Existential Deposit (ED; the
+/// pallet enforces that) to make sure that even if the message cost is below ED it is still payed
+/// to the relayer account.
+/// NOTE It's within relayer's interest to keep their balance above ED as well, to make sure they
+/// can receive the payment.
+pub struct InstantCurrencyPayments<T: frame_system::Config, Currency, RootAccount> {
+	_phantom: sp_std::marker::PhantomData<(T, Currency, RootAccount)>,
 }
 
-impl<AccountId, Currency> MessageDeliveryAndDispatchPayment<AccountId, Currency::Balance>
-	for InstantCurrencyPayments<AccountId, Currency>
+impl<T, Currency, RootAccount> MessageDeliveryAndDispatchPayment<T::AccountId, Currency::Balance>
+	for InstantCurrencyPayments<T, Currency, RootAccount>
 where
-	Currency: CurrencyT<AccountId>,
-	AccountId: Debug + Default + Encode,
+	T: frame_system::Config,
+	Currency: CurrencyT<T::AccountId>,
+	RootAccount: Get<Option<T::AccountId>>,
 {
 	type Error = &'static str;
 
+	fn initialize(relayer_fund_account: &T::AccountId) -> usize {
+		assert!(
+			frame_system::Module::<T>::account_exists(relayer_fund_account),
+			"The relayer fund account ({:?}) must exist for the message lanes pallet to work correctly.",
+			relayer_fund_account,
+		);
+		1
+	}
+
 	fn pay_delivery_and_dispatch_fee(
-		submitter: &Sender<AccountId>,
+		submitter: &Sender<T::AccountId>,
 		fee: &Currency::Balance,
-		relayer_fund_account: &AccountId,
+		relayer_fund_account: &T::AccountId,
 	) -> Result<(), Self::Error> {
-		match submitter {
-			Sender::Signed(submitter) => {
-				Currency::transfer(submitter, relayer_fund_account, *fee, ExistenceRequirement::AllowDeath)
-					.map_err(Into::into)
-			}
-			Sender::Root => {
-				Err("Sending messages from Root account is not supported yet. See GitHub issue #559 for more.")
-			}
-			Sender::None => {
-				Err("Sending messages from None account is not supported yet. See GitHub issue #559 for more.")
-			}
-		}
+		let root_account = RootAccount::get();
+		let account = match submitter {
+			Sender::Signed(submitter) => submitter,
+			Sender::Root | Sender::None => root_account
+				.as_ref()
+				.ok_or("Sending messages using Root or None origin is disallowed.")?,
+		};
+
+		Currency::transfer(
+			account,
+			relayer_fund_account,
+			*fee,
+			// it's fine for the submitter to go below Existential Deposit and die.
+			ExistenceRequirement::AllowDeath,
+		)
+		.map_err(Into::into)
 	}
 
 	fn pay_relayer_reward(
-		_confirmation_relayer: &AccountId,
-		relayer: &AccountId,
+		_confirmation_relayer: &T::AccountId,
```

Comment:
Could this parameter be removed from `MessageDeliveryAndDispatchPayment::pay_relayer_reward()`? AFAICT it as actually never used in any implementation of the trait. Or is this a placeholder for some future improvement?

Reply:
Yes, it's a placeholder, see #599. Or rather this one is a super simple implementation of the reward payment trait and we know that more sophisticated ones (which #599 implements) will need this.